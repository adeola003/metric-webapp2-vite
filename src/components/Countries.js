import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCountries, searchByRegion } from '../redux/countries/countriesSlice';

const Countries = () => {
  const { countriesData, loading, region } = useSelector((store) => store.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
    if (region) {
      dispatch(searchByRegion(region));
    }
  }, [dispatch, region]);

  return (
    <>
      {loading ? (
        <h1>Loading</h1>
      ) : (
        countriesData.map((country) => (
          <Link className="country-element" key={country.cioc} to={`/${country.cioc}`}>
            <div className="content">
              <img className="card-bg" src={country.flags.png} alt={`Flag of ${country.name.common}`} />
              <div className="overlay">
                <h3>{country.name.common}</h3>
                <p>
                  Region:&nbsp;
                  <span>{country.region}</span>
                </p>
                <p>
                  Capital:&nbsp;
                  <span>{country.capital}</span>
                </p>
              </div>
            </div>
          </Link>
        ))
      )}
    </>
  );
};

export default Countries;
