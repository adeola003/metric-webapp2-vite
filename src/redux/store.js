import { configureStore } from '@reduxjs/toolkit';
import { countriesReducer } from './countries/countriesSlice'; // Update the import statement

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
