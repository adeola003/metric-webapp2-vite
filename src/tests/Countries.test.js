import React from "react";
import renderer from "react-test-renderer";
import store from "../redux/store";
import Countries from "../components/Countries";
import { Provider } from "react-redux";


describe('Test Rockets component', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <Countries />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });