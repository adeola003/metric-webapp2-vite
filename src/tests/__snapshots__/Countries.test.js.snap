// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rockets component renders correctly 1`] = `null`;
